File: 11_03_2024_Custom_Sort_String.java
import java.util.HashMap;

class Solution {
    public String customSortString(String order, String s) {
        HashMap<Character, Integer> count = new HashMap<>();
        for (char c : order.toCharArray()) {
            count.put(c, 0);
        }
        for (char c : s.toCharArray()) {
            if (count.containsKey(c)) {
                count.put(c, count.get(c) + 1);
            }
        }
        StringBuilder ans = new StringBuilder();
        for (char c : order.toCharArray()) {
            ans.append(String.valueOf(c).repeat(count.get(c)));
        }
        for (char c : s.toCharArray()) {
            if (!count.containsKey(c)) {
                ans.append(c);
            }
        }
        return ans.toString();
    }
}

File: 11_03_2024_Custom_Sort_String.py
class Solution:
    def customSortString(self, order, s):
        Count = {}
        for c in order:
            Count[c] = 0
        for c in s:
            if c in Count:
                Count[c] += 1
        ans = ""
        for c in order:
            ans += Count[c] * c
        for c in s:
            if c not in Count:
                ans += c
        return ans

